fastplotlib/layouts/_gridplot.py

```diff
@@ -92,12 +92,35 @@ def __init__(
                 self.shape
             )
 
+        if "names" in kwargs.keys():
+            self.names = to_array(kwargs["names"])
+            print(self.names)
+            if self.names.shape != self.shape:
+                raise ValueError
+        else:
+            self.names = None
+
         if isinstance(controllers, str):
             if controllers == "sync":
                 controllers = np.zeros(
                     self.shape[0] * self.shape[1], dtype=int
                 ).reshape(self.shape)
 
+        if (self.names is not None) & all(isinstance(item, str) for item in controllers[0]):
+            idx_async_cont = np.shape(controllers)[0]
+            c = to_array(controllers)
+            controllers = np.zeros(
+                self.shape[0] * self.shape[1], dtype=int
+            ).reshape(self.shape)
+            for i in range(self.shape[0]):
```

Comment:
can just use `itertools.product` 

```
positions = list(product(range(self.shape[0]), range(self.shape[1])))
```

Reply:
I agree, this is quite nested.

btw, `list` is not necessary when using product, can just do `for positions in product(...)`

Reply:
merci beaucoup, changed.